.buildkite/pipeline.yml

```diff
@@ -0,0 +1,86 @@
+
+steps:
+- label: ":ruby: Rails Current"
+  timeout_in_minutes: 5
+  agents:
+    queue: ruby
+    ruby: 2.6.5
+  env:
+    JUNIT_OUTPUT_DIR: test/reports/current
+  commands:
+  - apt-get update && apt-get install -y libsqlite3-dev
+  - bundle install
+  - bundle exec rake db:migrate
+  - bundle exec rake test
+  artifact_paths:
+  - test/reports/**/*
+- label: ":ruby: Rails 4"
+  timeout_in_minutes: 5
+  agents:
+    queue: ruby
+    ruby: 2.6.5
+  env:
+    JUNIT_OUTPUT_DIR: test/reports/rails_4
+  commands:
+  - apt-get update && apt-get install -y libsqlite3-dev
+  - bundle install
+  - bundle exec appraisal install
+  - bundle exec rake db:migrate
+  - bundle exec appraisal rails-4 rake test
+  artifact_paths:
+  - test/reports/**/*
+- label: ":ruby: Rails 5"
+  timeout_in_minutes: 5
+  agents:
+    queue: ruby
+    ruby: 2.6.5
+  env:
+    JUNIT_OUTPUT_DIR: test/reports/rails_5
+  commands:
+  - apt-get update && apt-get install -y libsqlite3-dev
+  - bundle install
+  - bundle exec appraisal install
+  - bundle exec rake db:migrate
+  - bundle exec appraisal rails-5 rake test
```

Comment:
Used the `appraisal` gem in CI rather than specifying the `BUNDLE_GEMFILE` env var for the purpose of allowing the `db:migrate` step to run using activerecord v4.2 since. The error message `Directly inheriting from ActiveRecord::Migration is not supported` in rails 5+ is a familiar failure that we tackled in `declare_schema` but we currently don't have good sqlite3 support and don't depend on `declare_schema` in the aggregate gem

Reply:
If you use the following snippet to inherit the migration source, you can move away from using appraisal

`(ActiveSupport::VERSION::MAJOR >= 5 ? ActiveRecord::Migration[4.2] : ActiveRecord::Migration)`